src/Shared/E2ETesting/WaitAssert.cs

```diff
@@ -51,7 +54,7 @@ private static void WaitAssertCore(IWebDriver driver, Action assertion, TimeSpan
         {
             if (timeout == default)
             {
-                timeout = DefaultTimeout;
+                timeout = !TestRunFailed ? DefaultTimeout : FailureTimeout;
```

Comment:
This isn't a problem when multiple things start at once because we don't run E2E tests in parallel, correct?

Reply:
Nothing bad can happen from this, the worst case scenario is described below where the field is set.